app/controllers/posts_controller.rb

```diff
@@ -25,27 +26,39 @@ def show
 
 	def edit
 		@post = Post.find(params[:id])
+
+		if is_current_user_owner(@post)
```

Comment:
Looks like you need to introduce policies here, instead of this `if` block, take a look at this gem: https://github.com/elabs/pundit/